specification/0001-multifold.instance.json.md

```diff
@@ -0,0 +1,174 @@
+# multifold.instance.json
+
+## Summary
+
+This document acts as a Request for Comments (RPC) on the topic of how a MultiFold instance metadata is formatted.
+
+## Motivation
+
+This RFC describes a scalable, portable, and more powerful format for the instance that defines what an instance's
+definition should look like. MultiFold offers users the ability install mod loaders and mods seamlessly, without manual
+configuration.
+
+## Explanation
+
+The following is the schema for instance metadata. The primary format is JSON, however it can be represented in other
+formats such as YAML.
+
+<details>
+<summary>apiVersion: <code>string</code></summary>
+
+Represents the API version being used. Current supported API versions: `v1alpha1`.
+</details>
+
+<details> 
+<summary>kind: <code>string</code></summary>
+
+Represents the kind of the API object being used. In this instance, `Instance`.
+</details>
+
+<details> 
+<summary>metadata: <code>InstanceMetadata</code></summary>
+
+Contains metadata for the current instance, such as display name and group name.
+</details>
+
+<details> 
+<summary>spec: <code>InstanceSpec</code></summary>
+
+Contains spec for the current instance, such as Minecraft version, mod loader, and mods.
+</details>
+
+### InstanceMetadata
+
+<details> 
+<summary>name: <code>string</code></summary>
+
+The display name to use for the instance.
+</details>
+
+<details> 
+<summary>group: <code>string?</code></summary>
+
+The group name to group the instance with.
+</details>
+
+### InstanceSpec
+
+<details> 
+<summary>components: <code>InstanceComponent[]</code></summary>
+
+Array of "components" in the instance. This is used to describe the launching pipeline, and is order-sensitive.
+</details>
+
+<details> 
+<summary>mods: <code>InstanceMod[]</code></summary>
+
+Array of mods in the instance. This is used to automatically update mods in the instance and compatibility checks.
+</details>
+
+## InstanceComponent
+
+<details> 
+<summary>id: <code>string</code></summary>
+
+The id of the component. This must match a component in MultiFold, otherwise the instance will not launch.
+</details>
+
+<details> 
+<summary>version: <code>string</code></summary>
+
+The required version of the component. It is recommended to follow [semver](https://semver.org/) specs. It should be
+possible to define a "version range", similar to dependencies in `package.json`. Read more about version
+ranges [here](https://docs.npmjs.com/cli/v6/using-npm/semver).
+</details>
+
+<details> 
+<summary>settings: <code>Record&lt;string, string&gt;?</code></summary>
+
+Record or a map of settings. This is optional and can be freely defined. The options will be passed to the component
+upon initialization.
+</details>
+
+## InstanceMod
+
+<details> 
+<summary>id: <code>string</code></summary>
+
+The id of the mod **from the provider**. This is not to be confused with the mod id defined in mcmod.info or
```

Comment:
Explain further. Would this be for example the CurseForge mod ID in the URL or something else?